tests/baselines/reference/assignmentToObjectAndFunction.errors.txt

```diff
@@ -14,6 +16,7 @@ tests/cases/compiler/assignmentToObjectAndFunction.ts(29,5): error TS2322: Type
 !!! error TS2322: Type '{ toString: number; }' is not assignable to type 'Object'.
 !!! error TS2322:   Types of property 'toString' are incompatible.
 !!! error TS2322:     Type 'number' is not assignable to type '() => string'.
+!!! error TS2322:       Type 'Number' provides no match for the signature '(): string'

```

Comment:
Ah shucks, you need to check the type of the source *prior* to getting the apparent type. This might be more trouble than it's worth.